test/extended/util/framework.go

```diff
@@ -238,10 +238,56 @@ func processScanError(log string) error {
 	return fmt.Errorf(log)
 }
 
+// WaitForImageStreams creates custom ruby imageStream in OpenShift namespace
+// TODO: To eliminate the dependency on OpenShift Samples Operator in future,
+// WaitForImageStreams should be a replacement of WaitForOpenShiftNamespaceImageStreams func
+func WaitForImageStreams(oc *CLI) error {
+	ctx := context.Background()
+	var registryHostname string
+
+	images := map[string]string{
+		"ruby": "registry.access.redhat.com/ubi8/ruby-33:latest",
+	}
+
+	// Check to see if we have ImageRegistry enabled
+	hasImageRegistry, err := IsCapabilityEnabled(oc, configv1.ClusterVersionCapabilityImageRegistry)
+	if err != nil {
+		return err
+	}
+	if hasImageRegistry {
+		registryHostname, err = WaitForInternalRegistryHostname(oc)
+		if err != nil {
+			return err
+		}
+	}
+
+	// Create custom imageStream using `oc import-image`
+	e2e.Logf("waiting for imagestreams to be imported")
+	for image, location := range images {
+		err := oc.Run("import-image").Args(image, fmt.Sprintf("--from=%s", location), "--import-mode=PreserveOriginal", "--confirm").Execute()
```

Comment:
Won't this create an imagestream in oc's current namespace? If that is the case, the check for the openshift namespace imagestream will fail once we turn off the samples operator.